WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,164 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using Nito.AsyncEx;
+using Avalonia.Platform;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		private const byte DefaultCameraId = 0;
+
+		public WebcamQrReader(Network network)
+		{
+			ScanningTaskLock = new();
+			Network = network;
+		}
+
+		public event EventHandler<WriteableBitmap>? NewImageArrived;
+
+		public event EventHandler<string>? CorrectAddressFound;
+
+		public event EventHandler<string>? InvalidAddressFound;
+
+		public event EventHandler<Exception>? ErrorOccured;
+
+		private AsyncLock ScanningTaskLock { get; set; }
```

Comment:
Setter should not be here. You can initialize immediately here too, if you like:

```suggestion
		private AsyncLock ScanningTaskLock { get; } = new();
```

If you do like it, then the assignment should be removed from constructor.